targets/csharp/source/PlayFabSDK/WindowsFormsApp1/Form1.cs

```diff
@@ -0,0 +1,115 @@
+            await new PlayFabUtil.SynchronizationContextRemover();
+
+            var testInputs = GetTestTitleData(args);
+            UUnitIncrementalTestRunner.Start(true, null, testInputs, OnComplete);
+            while (!UUnitIncrementalTestRunner.SuiteFinished)
+            {
+                label1.Text = await UUnitIncrementalTestRunner.Tick();
+            }
+
+            WriteConsoleColor(UUnitIncrementalTestRunner.Summary);
+
+            return UUnitIncrementalTestRunner.AllTestsPassed ? 0 : 1;
+        }
+
+        private static void WriteConsoleColor(string msg = null, ConsoleColor textColor = ConsoleColor.White)
+        {
+            Console.ForegroundColor = textColor;
+            if (!string.IsNullOrEmpty(msg))
+                Console.WriteLine(msg);
+            Console.ForegroundColor = ConsoleColor.White;
+        }
+
+        private static int Pause(int code)
+        {
+            WriteConsoleColor("Done! Press any key to close", code == 0 ? ConsoleColor.Green : ConsoleColor.Red);
+            try
+            {
+                Console.ReadKey();
+            }
+            catch (InvalidOperationException)
+            {
+                // ReadKey fails when run from inside of Jenkins, so just ignore it.
+            }
+            return code;
+        }
+
+        private static TestTitleData GetTestTitleData(string[] args)
+        {
+            TestTitleData testInputs = null;
+            string filename = null;
+            for (var i = 0; i < args.Length; i++)
+                if (args[i] == "-testInputsFile" && (i + 1) < args.Length)
+                    filename = args[i + 1];
+            if (string.IsNullOrEmpty(filename))
+                filename = Environment.GetEnvironmentVariable("PF_TEST_TITLE_DATA_JSON");
+            if (File.Exists(filename))
+            {
+                var testInputsFile = File.ReadAllText(filename);
+                testInputs = PluginManager.GetPlugin<ISerializerPlugin>(PluginContract.PlayFab_Serializer).DeserializeObject<TestTitleData>(testInputsFile);
+            }
+            else
+            {
+                WriteConsoleColor("Loading testSettings file failed: " + filename, ConsoleColor.Red);
+                WriteConsoleColor("From: " + Directory.GetCurrentDirectory(), ConsoleColor.Red);
+            }
+            return testInputs;
+        }
+
+        private static void OnComplete(PlayFabResult<ExecuteCloudScriptResult> result)
+        {
+            WriteConsoleColor("Save to CloudScript result for: " + PlayFabSettings.BuildIdentifier + " => " + UUnitIncrementalTestRunner.PfClient.authenticationContext.PlayFabId, ConsoleColor.Gray);
+            if (result.Error != null)
+                WriteConsoleColor(result.Error.GenerateErrorReport(), ConsoleColor.Red);
+            else if (result.Result != null)
+                WriteConsoleColor("Successful!", ConsoleColor.Green);
+        }
+
+        private void button1_Click(object sender, EventArgs e)
```

Comment:
I'm assuming you have a button on your form then? At first glance it seems like an unneeded button callback

Reply:
Actually, it's necessary. It kicks off the test suite.
To repro the bug, I needed the entire PlayFab test suite to be executed directly off of a button in the most "inappropriate" way, to demonstrate that even if a customer does it utterly wrong, our lib won't deadlock their app.